web/main/tasks.py

```diff
@@ -1,50 +1,19 @@
-from typing import Optional, Union
-
 from django.conf import settings
-from django.core.mail import EmailMultiAlternatives
-from django.template import loader
-from django.utils.translation import activate
 
-from main.decorators import smtp_shell
+from main.services import CeleryProductsService
+from django.core.cache import cache
 from src.celery import app
 
 
-@app.task(name='email.send_information_email')
-def send_information_email(
-    subject: str,
-    template_name: str,
-    context: dict,
-    to_email: Union[list[str], str],
-    letter_language: str = 'en',
-    **kwargs: Optional[any],
-) -> bool:
-    """
-    :param subject: email subject
-    :param template_name: template path to email template
-    :param context: data what will be passed into email
-    :param to_email: receiver email(s)
-    :param letter_language: translate letter to selected lang
-    :param kwargs: from_email, bcc, cc, reply_to and file_path params
-    """
-    activate(letter_language)
-    to_email: list = [to_email] if isinstance(to_email, str) else to_email
-    email_message = EmailMultiAlternatives(
-        subject=subject,
-        from_email=kwargs.get('from_email'),
-        to=to_email,
-        bcc=kwargs.get('bcc'),
-        cc=kwargs.get('cc'),
-        reply_to=kwargs.get('reply_to'),
+@app.task
+def cache_active_channels():
+    if cache.keys(search=f'*{settings.CHANNEL_SETTINGS["CACHE_ACTIVE_CHANNELS_KEY"]}'):
+        return
```

Comment:
you need to force caching

Reply:
Okay, I will remove this condition and let the task purely cache every 20 minutes. But the reasons I had is to check whether we have a cache than we could just skip caching!